### PR TITLE
Fixed the auth flow for authorization with user context

### PR DESCRIPTION
**Pull request by dmeijboom**

### CODE REVIEW

File: api/inbox.go [hunk 1]
```diff
@@ -16,14 +17,40 @@ func (w Wrapper) GetTransferRequest(ctx echo.Context, requestorDID string, fhirT
 	if err != nil {
 		return err
 	}
-	transferRequest, err := w.TransferReceiverService.GetTransferRequest(ctx.Request().Context(), cid, requestorDID, fhirTaskID)
+
+	var sessionWithUserContext *Session
+
+	for _, session := range w.APIAuth.GetSessions() {
```
Comment (stevenvegt) [inline api/inbox.go:23]: Ik looks like you just get the fist session with a user context? When multiple people are logged in with an elevated session, don't you rist mixing them up?

Comment (dmeijboom) [reply api/inbox.go:23]: Ah yes indeed. Will fix ;)

File: api/auth.go [hunk 2]
```diff
@@ -65,7 +66,20 @@ func (auth *Auth) CreateCustomerJWT(customerId int) ([]byte, error) {
 	return jwt.Sign(t, jwa.ES256, auth.sessionKey)
 }
 
-func (auth Auth) GetCustomerIDFromHeader(ctx echo.Context) (int, error) {
+func (auth *Auth) GetSessions() map[string]Session {
```
Comment (reinkrul) [inline api/auth.go:69]: note that `Session.Credential` might not get copied, because it's an `interface{}`

Comment (reinkrul) [inline api/auth.go:69]: what about expired sessions?

Comment (dmeijboom) [reply api/auth.go:69]: Currently, we don't deal with expires sessions. You have to make sure to do everything within 15 minutes 😅

Comment (dmeijboom) [reply api/auth.go:69]: (that was already the case)

Comment (dmeijboom) [reply api/auth.go:69]: Indeed. I'm not sure if there is a generic way to (deep?) copy any data. This won't be an issue at runtime as we only use this to get a base64 encoded verifiable presentation which is a string.

**Review by reinkrul — APPROVED** [review]

**Pull request merged**